### ISSUE TITLE
space-infix-ops reports polymorphic functions

### ISSUE THREAD

**Issue by danez** [babel/babel-eslint#366]
``` js
const a = <T> (): void => {}
```

```
1:11  warning  Infix operators must be spaced  space-infix-ops
```

Follow up of #226

**Comment by Gozala** [comment]
Could this be related to https://phabricator.babeljs.io/T7330 which was fixed recently ?

**Comment by Gozala** [comment]
Given that babel-eslint seems to still be on [babylon@^6.0.18](https://github.com/babel/babel-eslint/blob/326d181520d97f73d4c65cf9ead7c351c21b268b/package.json#L17) it seems likely to me that update of babylon would fix this.

**Comment by hzoo** [comment]
we use `^` thought so you would just npm install

**Comment by Gozala** [comment]
> we use ^ thought so you would just npm install

Oh never mind then, I assumed it was only updating minor version. So I guess there is in fact some issue here.

**Comment by SIGAN** [comment]
The issue still **blocks** development with polymorphic static class method:
`cache.parse<T>('some argument')`

eslint 3.3.1
babel-eslint 6.1.2
eslint-plugin-flowtype 2.11.4

**Comment by remhume** [comment]
The issue still exists when calling any polymorphic functions/method with a given parameter like so:
`polyFunction<string>('hi');`

This is a bit frustrating, as flowtype is going in a direction of less magic inference (* is deprecated), and sometimes it is now required to give Polymorphic function parameters in complex situations or flow will report an error. 
```typescript
const fx = <T>(...) => {};

fx('foo'); // flow error: "Missing type annotation for `T`"

fx<string>('foo'); // no flow error, eslint error: "Infix operators must be spaced  space-infix-ops"
```

eslint: 5.2.0 (latest)
babel-eslint: 8.2.6 (latest)
eslint-plugin-flowtype: 2.50.0 (latest)
flow: 0.79 (latest)

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
